### PR TITLE
Moving generator related classes to TuistGenerator

### PR DESCRIPTION
**Pull request by kwridan**
Resolves https://github.com/tuist/tuist/issues/205

### Short description 📝

Tuist is a command line tool that offers several useful features related to managing Xcode projects at scale. The generation features are worthy of bundling in a library to enable external tools to leverage it and build on top of it. 

### Solution 📦

The `TuistGenerator` library has been previously created and several changes have been incrementally made to allow seamlessly migrating all the generation related code to it.

### Implementation 👩‍💻👨‍💻

- [x] Move sources to `TuistGenerator`
- [x] Move tests to `TuistGeneratorTests`
- [x] Update imports

### Test Plan ⚒

- Ensure unit tests continue to pass
  `swift test`

- Ensure acceptance tests continue to pass
  `bundle exec rake features`

- Manually generate any of the projects within `fixtures`
   `tuist generate`
  
### Notes 📝

`TuistGenerator` is not yet stable - breaking changes will most likely be made.

### CODE REVIEW

**Comment by kwridan** [comment]
Submitting a draft PR for us to take a look - If anyone spots candidates for smaller PRs please do suggest them 🙂 

Also happy for us to wait till the next release before merging this if we're planning on making one soon.

File: Sources/TuistGenerator/Models/Scheme.swift [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import Basic
+import Foundation
+
+public class Scheme: Equatable {
```
Comment (kwridan) [inline Sources/TuistGenerator/Models/Scheme.swift]: It's a shame - git didn't recognize this as a moved file

Comment (ollieatkinson) [reply Sources/TuistGenerator/Models/Scheme.swift]: If the code inside has changed git will struggle to keep the reference, you could give it a helping hand and use `git mv` [ref](https://git-scm.com/docs/git-mv)

**Comment by pepibumur** [comment]
@kwridan would you mind leaving some comments on the files that are not moves? It'll make the review easier 😅

**Comment by pepibumur** [comment]
Ouch, sorry. Clicked the wrong button and closed the PR.

File: Sources/TuistGenerator/Generator.swift [hunk 2]
```diff
@@ -1,62 +1,108 @@
 import Basic
+import Foundation
 import TuistCore
 
+public struct GeneratorConfig {
```
Comment (kwridan) [inline Sources/TuistGenerator/Generator.swift]: This is effectively a move from `TuistKit/Generator/Generator.swift` > `TuistGenerator/Generator.swift`

`TuistKit/Generator/Generator.swift` only now hosts the extension that deals with the manifest - we could beef it up to its own class too

File: Sources/TuistKit/Generator/Generator.swift [hunk 3]
```diff
@@ -1,24 +1,5 @@
...
-    }
-}
-
-protocol Generating {
-    func generateProject(at path: AbsolutePath, config: GeneratorConfig, workspaceFiles: [AbsolutePath]) throws -> AbsolutePath
-    func generateWorkspace(at path: AbsolutePath, config: GeneratorConfig, workspaceFiles: [AbsolutePath]) throws -> AbsolutePath
-}
+import TuistGenerator
 
 extension Generating {
```
Comment (kwridan) [inline Sources/TuistKit/Generator/Generator.swift]: Only the extension is left here - we could beef this up to a class if needed

File: Tests/TuistGeneratorTests/GeneratorTests.swift [hunk 4]
```diff
@@ -1,35 +1,152 @@
 import Basic
 import XCTest
+@testable import TuistCoreTesting
```
Comment (kwridan) [inline Tests/TuistGeneratorTests/GeneratorTests.swift]: Similarly this is a move (we had a placeholder test file previously hence why it's highlighting some changes)

File: Tests/TuistGeneratorTests/Utils/AbsolutePath+String.swift [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import Basic
+
+extension AbsolutePath: ExpressibleByStringLiteral {
```
Comment (kwridan) [inline Tests/TuistGeneratorTests/Utils/AbsolutePath+String.swift]: This was copied - as a few tests translate Strings > AbsolutePath for convenience

File: Sources/TuistKit/Generator/GeneratorModelLoader.swift [hunk 6]
```diff
@@ -6,17 +6,22 @@ import TuistGenerator
 
 enum GeneratorModelLoaderError: Error, Equatable, FatalError {
     case featureNotYetSupported(String)
+    case missingFile(AbsolutePath)
```
Comment (kwridan) [inline Sources/TuistKit/Generator/GeneratorModelLoader.swift]: This was added - 

Previously this class re-used `GraphLoaderError` to throw this kind of error :/

File: Sources/TuistCore/Linter/LintingIssue.swift [hunk 7]
```diff
@@ -1,47 +1,46 @@
 import Foundation
-import TuistCore
 
-struct LintingError: FatalError, Equatable {
-    var description: String = "Fatal linting issues found"
-    var type: ErrorType = .abort
+public struct LintingError: FatalError, Equatable {
+    public var description: String = "Fatal linting issues found"
```
Comment (kwridan) [inline Sources/TuistCore/Linter/LintingIssue.swift]: This was moved to `TuistCore` to allow both modules to use it

**Pull request merged**